compair/static/modules/session/session-service.js

```diff
@@ -125,14 +157,82 @@
                     angular.extend(scope._user, u);
                     scope._permissions = result.data.permissions;
                     localStorageService.set('permissions', scope._permissions);
+                    scope._impersonation = result.data.impersonation;
+                    localStorageService.set('impersonation', scope._impersonation);
                     return deferred.promise;
                 });
             },
             expirePermissions: function() {
                 this._permissions = null;
                 localStorageService.remove('permissions');
-            }
+            },
+
+            getImpersonation: function() {
```

Comment:
This is taking similar approach as `getPermissions`.  The `/api/session` also piggybacked the permission information.  But I guess for performance consideration, if the permission info is not found in cache, instead of retrieving the whole session info via `/api/session`, `getPermissions` involves `/api/session/permission`.